src/main/java/org/springframework/data/couchbase/core/ExecutableFindByAnalyticsOperationSupport.java

```diff
@@ -53,7 +53,7 @@ public ExecutableFindByAnalyticsOperationSupport(final CouchbaseTemplate templat
 			this.domainType = domainType;
 			this.query = query;
 			this.reactiveSupport = new ReactiveFindByAnalyticsSupport<>(template.reactive(), domainType, query,
-					scanConsistency);
+					scanConsistency, new NonReactiveSupportWrapper(template.support()));
```

Comment:
Isn't this NonReactiveSupportWrapper object the same as template.reactive().support() ? 
Then this extra argument  to the Reactive*Support constructors is not necessary.
And the ReactiveTemplateSupport support object in the Reactive*Support objects is not necessary - they can just use template.support().

Reply:
Hi @mikereiche 

In my opinion and following the implementations of other spring-data projects such as spring-data-mongo, I believe that in  blocked stack we should launch `EntityCallback` and from the reactive stack we should launch `ReactiveEntityCallback`.
To have that differentiation, there needs to be one path for blocking (`CouchbaseTemplateSupport`) and another path for reactive (`ReactiveCouchbaseTemplateSupport`).

Also, I think that, additionally, the blocking audit event should be linked to `AuditingEntityCallback` instead of `AuditingEventListener` (this is not in the PR but you could add it).

How do you see it?

Reply:
Got it. Thanks for the explanation.

Reply:
This point
> Also, I think that, additionally, the blocking audit event should be linked to AuditingEntityCallback instead of AuditingEventListener (this is not in the PR but you could add it).

Do you want me to add it to this PR?

Reply:
Yes please.

Reply:
Done